api/envoy/extensions/filters/http/jwt_authn/v3/config.proto

```diff
@@ -60,10 +60,25 @@ message JwtProvider {
   // Specify the `principal <https://tools.ietf.org/html/rfc7519#section-4.1.1>`_ that issued
   // the JWT, usually a URL or an email address.
   //
+  // It is optional. If specified, it has to match the *iss* field in JWT.
+  //
+  // If a JWT has *iss* field and this field is specified, they have to match, otherwise the
+  // JWT *iss* field is not checked.
+  //
+  // If *allow_missing* or *allow_missing_or_failed* is used, the JWT *iss* field is used to find
```

Comment:
If the JWT is missing (with `allow_missing`), where do you get the JWT *iss* field from?

Reply:
This is for the case where if there is a JWT, how to handle it.

Reply:
Maybe I'm a bit confused then; why does `allow_missing` have special semantics when the JWT is not missing?

Reply:
with "allow_missing",  if there is a JWT token in the request,  we still have to verify it and if verification fails,  the request is rejected.  But if JWT is missing,  it is OK, the request is proceeded.   This is used for "optional" JWT verification usage.

Reply:
I guess my previous comment did not answer your question.  Why semantics is different in `allow_missing` case?

For supporting backward compatibility, 'allowing_missing` and `allow_failed` is implemented differently than the other normal cases.   In other normal cases,  it is against a specific `provider`, not need to use "iss" to find the "provider".  But in the former case,  it is against all providers,  we have to use "iss" from JWT to find the provider to verify.  That is why the semantic different.

Reply:
Makes sense. Could you add a `[#comment: ]` explaining this for future folks who are trying to understand? I appreciate your patience here!

Reply:
Done